pkgs/games/prismlauncher/wrapper.nix

```diff
@@ -21,71 +24,83 @@
 , msaClientID ? null
 , gamemodeSupport ? stdenv.isLinux
 , textToSpeechSupport ? stdenv.isLinux
+, waylandSupport ? false
+, shellWrapper ? waylandSupport
 , jdks ? [ jdk17 jdk8 ]
 , additionalLibs ? [ ]
 , additionalPrograms ? [ ]
 }:
+
+assert lib.assertMsg (waylandSupport -> stdenv.isLinux) "waylandSupport is only available on Linux";
+assert lib.assertMsg (waylandSupport -> shellWrapper) "waylandSupport requires shellWrapper";
+
 let
-  prismlauncherFinal = prismlauncher-unwrapped.override {
+  super = prismlauncher-unwrapped.override {
```

Comment:
i think `final` would be a better name here, as it doesn't conflict with something in nixpkgs scope and is more representative of it being the *final* package after being overriden

Reply:
`final` is a bad name, because it is not final. In the context of the wrapper, we are evaluating what would be `final` or `self`.

`super` is the most accurate name, because it's the package we're wrapping. `packageSuper` is also another accurate name.

Which of these options do you like best?

- `packageSuper`
- `package'`
- `prismlauncher'`
- `unwrappedPackage`

Reply:
I would argue `final` `prev` `super` and `self` are all bad choices as those are used in the context of overlays and overrides, so the meaning wouldn't translate well here.

`prismlauncher-unwrapped'` is fine, I guess

Reply:
> In the context of the wrapper, we are evaluating what would be final or self.

yes for `prismlauncher`, but not for `prismlauncher-unwrapped`. the post-override version of it we use is the `final` of `prismlauncher-unwrapped` in the context of this expression - just as it would be in an overlay for example. i can see why there could be confusion and debate over this though, so maybe changing it to `prismlauncher-unwrapped'` would be good. it'd make it more apparent what exactly in nixpkgs we're referencing, while also being a pretty unopinionated name

Reply:
I vote for `prismlauncher'` because it is shorter, and mentioning that the package is "unwrapped" is unnecessary considering all three usages of the identifier are close together, so it's easy to see the `prismlauncher' = prismlauncher-unwrapped.override`. 

Actually, because of that, I still stand by `package'` because there's only one package being derived here. I isually keep the "main" package as `package'` and name the others by their full name + `'`.

For now, latest commit still uses `prismlauncherFinal`.